modular_chomp/code/modules/vote/vote_datum.dm

```diff
@@ -0,0 +1,200 @@
+/datum/vote
+	// Person who started the vote
+	var/initiator = "the server"
+	// world.time the bote started at
+	var/started_time
+	// The question being asked
+	var/question
+	// Vote type text, for showing in UIs and stuff
+	var/vote_type_text = "unset"
+	// Do we want to show the vote count as it goes
+	var/show_counts = FALSE
+	// Vote result type. This determines how a winner is picked
+	var/vote_result_type = VOTE_RESULT_TYPE_MAJORITY
+	// Was this this vote custom started?
+	var/is_custom = FALSE
+	// Choices available in the vote
+	var/list/choices = list()
+	// Assoc list of [ckeys => choice] who have voted. We don't want to hold clients refs.___callbackvarset(list_or_datum, var_name, var_value)
+	var/list/voted = list()
+	// For how long will it be up
+	var/vote_time = 60
+
+/datum/vote/New(var/_initiator, var/_question, list/_choices, var/_is_custom = FALSE)
+	if(SSvote.active_vote)
+		CRASH("Attempted to start another vote with one already in progress!")
```

Comment:
Do we really crash when there's already a vote ongoing? Is it sanity checked to prevent this from normally happening?